main/src/main/scala/sbt/RemoteCache.scala

```diff
@@ -50,27 +50,6 @@ object RemoteCache {
   private[sbt] val analysisStore: mutable.Map[HashedVirtualFileRef, CompileAnalysis] =
     mutable.Map.empty
 
-  // TODO: figure out a good timing to initialize cache
-  // currently this is called twice so metabuild can call compile with a minimal setting
-  private[sbt] def initializeRemoteCache(s: State): Unit =
```

Comment:
The remote cache and the `BuildWideCacheConfiguration` are now initialized by the task engine (as a task but we can make the `CacheStore` a setting if we need to persist the instance).